### PR TITLE
Fix network cli prompts to bind to the newline

### PR DESCRIPTION
**Pull request by bdowling**
##### SUMMARY

This is a follow-on to #34585 where these other network_cli regexps should also be changed.  This PR changes the ones mentioned in https://github.com/ansible/ansible/pull/34585#issuecomment-356788884 for the password: prompts, and a second commit to this branch changes the normal CLI prompts as well.

Essentially all the prompt and password sub-prompts should be bound to the start of a newline.  Making it optional allows greater freedom in finding these in other content network devices can emit.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/plugins/terminal/enos.py
lib/ansible/plugins/terminal/eos.py
lib/ansible/plugins/terminal/dellos10.py
lib/ansible/plugins/terminal/asa.py
lib/ansible/plugins/terminal/ironware.py
lib/ansible/plugins/terminal/dellos9.py
lib/ansible/plugins/terminal/dellos6.py
lib/ansible/module_utils/network/dellos9/dellos9.py
lib/ansible/module_utils/network/dellos9/dellos9.py
lib/ansible/module_utils/network/dellos9/dellos9.py
lib/ansible/module_utils/network/dellos6/dellos6.py
lib/ansible/module_utils/network/dellos6/dellos6.py
lib/ansible/module_utils/network/dellos6/dellos6.py
lib/ansible/module_utils/network/dellos10/dellos10.py
lib/ansible/module_utils/network/dellos10/dellos10.py
lib/ansible/module_utils/network/dellos10/dellos10.py
lib/ansible/plugins/terminal/onyx.py

##### ANSIBLE VERSION

```
devel
```

### CODE REVIEW

**Comment by bdowling** [comment]
Also, because #34723 was also merged, these changes should follow closely along or others may encounter issues with the mentioned modules.

**Review by trishnaguha — DISMISSED** [review]

**Review by daxlab — APPROVED** [review]

**Comment by ganeshrn** [comment]
Since this PR touches multiple platforms, it needs approval from individual platform maintainers
/cc @ogenstad @skg-net @amuraleedhar @paulquack

**Comment by samerd** [comment]
works fine with Mellanox onyx

**Review by Qalthos — APPROVED** [review]

**Review by paulquack — APPROVED** [review]
Works on ironware

File: lib/ansible/plugins/terminal/ironware.py
Comment (paulquack) [inline lib/ansible/plugins/terminal/ironware.py:33]: Actually this doesn't seem to work for ironware, not sure exactly why not though. 

The full traceback is:
Traceback (most recent call last):
  File "/Users/paul.baker/ansible/bin/ansible-connection", line 70, in start
    self.connection._connect()
  File "/Users/paul.baker/ansible/lib/ansible/plugins/connection/network_cli.py", line 315, in _connect
    self.receive()
  File "/Users/paul.baker/ansible/lib/ansible/plugins/connection/network_cli.py", line 387, in receive
    data = self._ssh_shell.recv(256)
  File "/usr/local/lib/python2.7/site-packages/paramiko/channel.py", line 685, in recv
    raise socket.timeout()

Could it be that there is no newline character when we first open a connection?

When I SSH to the router this is what I see, and this does seem to match the RE above.
$ ssh paulb@rtr02.mel01
Password:
SSH@rtr02.mel01#

**Review by paulquack — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/terminal/ironware.py
Comment (bdowling) [reply lib/ansible/plugins/terminal/ironware.py:33]: @paulquack Did you happen to dig into this anymore, it definitely looks like that regex would match.  Are there any other non-printing characters before the prompt?  A bell `^G` or anything?

**Comment by Qalthos** [comment]
So this change has been in merge conflict limbo for months. More importantly, however, the original commit has now already been reverted (#39063). As such, I'm closing this, and any future change of this should probably be looked at case-by-case to make sure similar issues are not possible on that platform.

**Pull request closed without merging**
